src/android/io/sqlc/SQLitePlugin.java

```diff
@@ -256,7 +264,15 @@ private SQLiteAndroidDatabase openDatabase(String dbname, String key, CallbackCo
             return mydb;
         } catch (Exception e) {
             // NOTE: NO Android locking/closing BUG workaround needed here
-            cbc.error("can't open database " + e);
+
+            if(selfHealingEnabled && e.getMessage().contains("file is encrypted or is not a database:")) {
+                logger.logError("Android database will be deleted to self heal:" + e.getMessage(), "SQLite");
```

Comment:
Can we have a better error message here? Is it possible to get the reason why this happened?

Reply:
Doesn't the "e.getMessage()" contain the reason?

Reply:
You tell me 😄 I have no context about these error messages 😞

Reply:
Ahhh ok... Yes, the getMessage contains the reason behind the error 😄 If this error appears again, we will be able to understand the reason.